python/proton/_message.py

```diff
@@ -515,18 +515,20 @@ def _set_annotations(self, annotations):
 
     def encode(self):
         self._pre_encode()
+        bytes = ffi.new("char *")
         sz = 16
+        size = ffi.new("size_t *", sz)
         while True:
-            err, data = pn_message_encode(self._msg, sz)
-            if err == PN_OVERFLOW:
+            data = pn_message_encode(self._msg, bytes, size)
```

Comment:
I'm not sure this approach is correct. Test cases are failing without trace

Reply:
Are the tests failing on your own machine in this way? When you have a timeout in the tests, you should figure out where is the test stuck. There are several options. I will tell you the simplest one, to start with:

Put `print("I am still alive, number of this print is 42")` calls into the test, put multiple calls like this, with different numbers. Find which is the last one that is printed and first one that is not printed.

Reply:
And yes, this is wrong. I don't understand why it is looping (I'd expect to see a crash out of this instead), but you can debug this using the prints at first, and with other techniques later, when you establish which line in python code is causing the looping by debugging using the prints.